contracts/interchain-token-service/src/contract.rs

```diff
@@ -365,7 +365,7 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
 
         let message = Message::InterchainTransfer(InterchainTransfer {
             token_id,
-            source_address: caller.clone().to_xdr(env),
+            source_address: caller.to_bytes(env),
```

Comment:
why didn't golden file change due to this? That should be updated as well. Or modify the tests to ensure this is covered